### PR TITLE
feat: use PluginSettings components for Campaigns wizard

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Refactors the **Campaigns > Settings** wizard page to use the `PluginSettings` component. Requires https://github.com/Automattic/newspack-popups/pull/777.

### How to test the changes in this Pull Request:

1. Make sure your Campaigns repo is checked out with https://github.com/Automattic/newspack-popups/pull/777.
2. Visit **Newspack > Campaigns > Settings**.
3. Confirm you see the new settings UI:

Screen Shot 2022-01-21 at 1 49 18 PM

4. Change various settings and save, then refresh the page. Confirm that settings are saved as expected.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/components/src/plugin-settings/SettingsSection.js
Comment (miguelpeixe) [inline assets/components/src/plugin-settings/SettingsSection.js:30]: This case is not necessary. The use of `<SelectControl />` is determined by the existence of `setting.options`:

```js
	if ( Array.isArray( setting.options ) && setting.options.length ) {
		return SelectControl;
	}
```

It allows for the setting to preserve type validation.

File: assets/components/src/plugin-settings/SettingsSection.js
Comment (miguelpeixe) [inline assets/components/src/plugin-settings/SettingsSection.js:51]: This is also not necessary since `SelectControl` is a `<select />` element, which doesn't accept the `type` attribute.

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
The settings are working well. I feel that the grouping of multiple checkboxes looks a bit cluttered. Maybe an adjustment of the `help` font size or margin could help? cc @thomasguillot

File: assets/components/src/plugin-settings/SettingsSection.js
Comment (dkoo) [reply assets/components/src/plugin-settings/SettingsSection.js]: Fixed in 4eef95d6

File: assets/components/src/plugin-settings/SettingsSection.js
Comment (dkoo) [reply assets/components/src/plugin-settings/SettingsSection.js]: Thanks for clarifying! Fixed in 4eef95d6

**Comment by dkoo** [comment]
> I feel that the grouping of multiple checkboxes looks a bit cluttered. Maybe an adjustment of the help font size or margin could help?

I agree! Changing the font-size might help a bit, perhaps changing the color too? It still looks a bit off-kilter when the settings labels have different lengths, though:

Screen Shot 2022-01-25 at 10 19 20 AM

I'll defer to @thomasguillot on this, and I think this can be handled in a separate PR if needed too.

**Review by miguelpeixe — APPROVED** [review]
Thank you for the revisions!

I agree it can be handled in a different PR. I already see improvement on your screenshot and believe the difference in label length could be resolved by vertically aligning the help text to the label. We also need to see how that works with other controls' help texts.

**Comment by dkoo** [comment]
Thanks for reviewing—I'll hold off on merging until https://github.com/Automattic/newspack-popups/pull/777 is ready to be merged as well.

**Comment by dkoo** [comment]
769d572a updates some method names used to update Campaigns settings sections to avoid confusion with that plugin's standalone/alternative settings UI's methods, based on [this suggestion](https://github.com/Automattic/newspack-popups/pull/777#discussion_r802116322) in Campaigns.

**Review by miguelpeixe — APPROVED** [review]

**Pull request merged**
